docs/design/server_side_charts.md

```diff
@@ -0,0 +1,44 @@
+# Supporting charts on the server side
+As discussed in #367, the server side components manager and resourcifier have not yet been refactored, as of this writing, to work with [the new chart format](docs/design/chart_format.md). Presumably, expandybird should not be affected.
+
+## Context
+In discussions about the refactoring on the Slack channel, a number of potential requirements were discussed that could complicate matters, such as:
+* supporting multiple templating technologies within a single chart.
+* supporting multiple templating technologies within a single dependency graph.
+* supporting the parameterization of charts, as opposed to templates.
```

Comment:
OK, but in that case, you'd need a way to bind the parameters passed to the chart to the various templates inside, which implies some kind of mapping, which again reduces the forest to a tree. The only difference is that the top level is now a map, instead of a template. Either way, the result is the same... There's a single top level thing that gets processed by some engine, which receives the incoming parameters and uses them in a well defined way.